### PR TITLE
Allow toxlsx() to overwrite add or replace a worksheet

### PR DESCRIPTION
**Pull request by juarezr**
This PR is the same as #502 but rebased on top of #506 

## Changes

1. Added new parameter `mode` to `toxlsx()` from $502
2. Rebased on top of #506
3. Removed a call to `os.path.exists()` as it wont work with remote sources from `fsspec`

## Checklist

* [x] Code
  * [x] Includes unit tests
  * [x] New functions have docstrings with examples that can be run with doctest
  * [x] New functions are included in API docs
  * [x] Docstrings include notes for any changes to API or behaviour
  * [ ] All changes documented in docs/changes.rst
* [x] Testing
  * [x] \(Optional) Tested local against remote servers
  * [x] Travis CI passes (unit tests run under Linux)
  * [x] AppVeyor CI passes (unit tests run under Windows)
  * [x] Unit test coverage has not decreased (see Coveralls)
* [x] Changes
  * [x] Ready to review
  * [x] Ready to merge

### CODE REVIEW

**Comment by juarezr** [comment]
@alimanfoo , @chrullrich , 

Everybody busy? :smile:

File: petl/test/io/test_remotes.py [hunk 1]
```diff
@@ -3,11 +3,16 @@
 
 import sys
 import os
+import time
```
Comment (chrullrich) [inline petl/test/io/test_remotes.py:6]: `time` is not used, is it?

File: petl/io/xlsx.py [hunk 2]
```diff
@@ -86,22 +86,80 @@ def __iter__(self):
...
+
+def _load_or_create_workbook(filename, mode, sheet):
+    if PY3:
+        FileNotFound = FileNotFoundError
+    else:
+        FileNotFound = IOError
+
+    import openpyxl
+    wb = None
+    if mode != "overwrite" and (mode != "replace" or sheet is not None):
```
[the remark this answers is not in the document]
Comment (juarezr) [reply petl/io/xlsx.py:134]: Let's use the easier to understand...

File: petl/test/io/test_remotes.py (hunk 1 above)
Comment (juarezr) [reply petl/test/io/test_remotes.py:6]: No.
Also flake8 was complaining about it!

File: petl/io/xlsx.py [hunk 3]
```diff
@@ -86,22 +86,80 @@ def __iter__(self):
...
+        try:
+            source = read_source_from_arg(filename)
+            with source.open('rb') as source2:
+                wb = openpyxl.load_workbook(filename=source2, read_only=False)
+        except FileNotFound:
+            wb = None
```
[the remark this answers is not in the document]
Comment (juarezr) [reply petl/io/xlsx.py:140]: If I have understood correctly, fixing the bug above should fix this behavior.

Right?

**Comment by juarezr** [comment]
Resuming this merge because:

1. The review issues are addressed.
2. There is no incompatible change in public API or existing behavior.
3. We can make further changes if a problem arises.
4. In the worst situation we can revert this back.
5. Everybody is too busy for being bothered re-reviewing.

**Pull request merged**